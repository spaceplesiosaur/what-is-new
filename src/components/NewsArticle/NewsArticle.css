.articleBlock {
  display: flex;
  flex-direction: column;
  width: 400px;
  margin: 10px;
}

.articlePic {
  height: 200px;
  width: 400px;
}

.articleHeader {
  background-color: white;
  font-weight: bold;
  height: 75px;
  padding: 1.5em 0 1em 0;
}

.articleDescription {
  background-color: white;
  height: 100px;
  overflow: scroll;
}

.articleUrl {
  background-color: lavender;
}
