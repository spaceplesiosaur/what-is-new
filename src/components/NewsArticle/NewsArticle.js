import React from 'react';
import './NewsArticle.css';

const NewsArticle = props => {
  return (<section className="articleBlock" id={props.id}>
    <img className="articlePic" alt="unique" src={props.img}/>
    <h1 className="articleHeader">{props.headline}</h1>
    <p className="articleDescription">{props.description}</p>
    <a href={props.url} className="articleUrl">See more...</a>
  </section>)
}

export default NewsArticle;
