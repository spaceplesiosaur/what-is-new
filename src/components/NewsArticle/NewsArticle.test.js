import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () => {
  it('should match the snapshot with correct data', () => {
    const wrapper = shallow(<NewsArticle
      key='test'
      id='test'
      img='test'
      headline='test'
      description='test'
      url='test'
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
