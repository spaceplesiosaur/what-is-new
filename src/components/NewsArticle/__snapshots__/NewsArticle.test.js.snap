// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot with correct data 1`] = `
<section
  className="articleBlock"
  id="test"
>
  <img
    alt="unique"
    className="articlePic"
    src="test"
  />
  <h1
    className="articleHeader"
  >
    test
  </h1>
  <p
    className="articleDescription"
  >
    test
  </p>
  <a
    className="articleUrl"
    href="test"
  >
    Link
  </a>
</section>
`;
